### ISSUE TITLE
Difference between nullableTimestamps() and timestamps() ?

### ISSUE THREAD

**Issue by ntopulos** [laravel/internals#874]
- Laravel Version: 5.5.20

### Description:

The [documention](https://github.com/laravel/docs/blob/5.5/migrations.md#creating-columns) describes two timestamps commands, that seem identical:

> Command  | Description
> ------------- | -------------
> `$table->nullableTimestamps();`  |  Adds nullable versions of `timestamps()` columns.
> `$table->timestamps();`  |  Adds nullable `created_at` and `updated_at` TIMESTAMP equivalent columns.

Is there an issue with the documentation? For instance, not describing the differences or the fact that one is a legacy command that will be soon removed.
Or is the first command useless and one simple forgot to remove it? In that case it would be an issue with the framework.

**Comment by drbyte** [comment]
According to the code comments, `nullableTimestamps()` is an alias for `timestamps()`.
See: https://github.com/laravel/framework/blob/5.5/src/Illuminate/Database/Schema/Blueprint.php#L845-L869

If you dig back into older versions, you'll see that `timestamps()` at one point didn't do the `nullable` part:
https://github.com/laravel/framework/blob/4.0/src/Illuminate/Database/Schema/Blueprint.php#L600-L622

So, it would appear that as the framework has evolved it became evident that making the `timestamps()` method add nullable was necessary, but in the interest of backward compatibility the (now redundant) `nullableTimestamps()` method has been retained. No indication has been given of any plans to officially remove it from the framework.

Your point is valid insomuch as there's probably no need to document the `nullableTimestamps()` method anymore, as it creates confusion, and delivers the same outcome. 
I suggest doing a PR to the docs.

**Comment by drbyte** [comment]
Laravel doesn't strictly follow semver.

But I'd be inclined to do a PR to remove the ambiguity now, by removing the `nullableTimestamps()` listing from the docs. Judging from the look of the docs I think the nullableTimestamps entry has simply been forgotten to be removed. Taylor may have more plans, but if he agrees with you he'll merge it.

As support material, in your comments, you can point to the commit you quoted above, and perhaps also to the line of code where the docblocks say it's an alias.

**Comment by sisve** [comment]
Just to clarify; Laravel doesn't follow semver at all.

But that doesn't matter. Database migrations have a much longer lifespan than other code. Removing the `Blueprint::nullableTimestamps` method will cause existing migrations to break. If we teach people to not go back and edit migrations after they've been pushed, isn't it weird to tell them to go back and edit them because we broke it?

**Comment by drbyte** [comment]
Agreed insomuch as not deprecating it formally anytime soon.

But the docs don't need to list both methods, as that encourages the use of the old alias instead of moving forward to the seemingly-preferred way.

**Comment by Dylan-DPC** [comment]
> f we teach people to not go back and edit migrations after they've been pushed, isn't it weird to tell them to go back and edit them because we broke it?

I dont' see the problem with this. Since the behaviour (using deprecated function before it is being deprecated) is same as the behaviour of changing it to the other function after depreciation, users don't have to rerun their migrations

**Comment by paulofreitas** [comment]
The `timestamps()` method got changed on Laravel 5.2.14 to use the `nullable()` modifier after this bug report: https://github.com/laravel/framework/issues/12060

This issue required an **IMMEDIATE ACTION** since this MySQL behavior wasn't expected.

From [MySQL documentation](https://dev.mysql.com/doc/refman/5.7/en/timestamp-initialization.html):

> TIMESTAMP and DATETIME columns have no automatic properties unless they are specified explicitly, with this exception: If the explicit_defaults_for_timestamp system variable is disabled, the first TIMESTAMP column has both DEFAULT CURRENT_TIMESTAMP and ON UPDATE CURRENT_TIMESTAMP if neither is specified explicitly.

Not changing this would also break on MySQL 5.7.4+ because strict mode got enabled: https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sql-mode-changes

**Without strict mode:**

If you use non-nullable `TIMESTAMP` columns, the first column will automatically get the `DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP` behavior and any other declared will get `DEFAULT '0000-00-00 00:00:00'`.

If you declare any non-nullable `timestamp()` column in your migration **before** declaring `timestamps()` for creation/update timestamps, your own timestamp column will default to `CURRENT_TIMESTAMP` and the creation/update timestamps will default to `0000-00-00 00:00:00`.

If you declare any non-nullable `timestamp()` column in your migration **after** declaring `timestamps()` for creation/update timestamps, the creation timestamp column will default to `CURRENT_TIMESTAMP` and the others will default to `0000-00-00 00:00:00`.

**With strict mode:** (enabled by default since MySQL 5.7.4)

If you use non-nullable `TIMESTAMP` columns, you are required to declare a default value or change the column to be nullable.

Changing `timestamps()` to use the `nullable()` modifier was not only required but also a critical fix - this kind of fix are allowed to break any promise. 👍 

I also think it's not necessary to remove or deprecate the `nullableTimestamps ()` method, instead what need to change is the documentation to state it as an alias instead of removing it from there - I'll handle that for you. 😉

**Comment by Dylan-DPC** [comment]
@paulofreitas taylor has said that he has no plans of deprecating or removing the method.

**Comment by paulofreitas** [comment]
@Dylan-DPC Yeah, I saw his response in your PR. I forgot to mention "As Taylor I also think [...]" in the comment above. 😜 

My whole comment was to just make clear why `timestamps()` method has suddenly changed its earlier behavior. 😉

**Comment by Dylan-DPC** [comment]
I know it has changed over time and ended up matching the other function. The problem is we have 2 functions that behave in the same way and could be misleading

**Comment by paulofreitas** [comment]
> The problem is we have 2 functions that behave in the same way and could be misleading

Not anymore: https://github.com/laravel/docs/pull/3860 😄

**Comment by Dylan-DPC** [comment]
Ah cool that makes it better but doesn't solve the problem completely. We still have 2 functions for 1 task.
